crates/net/rpc/src/web3.rs

```diff
@@ -0,0 +1,41 @@
+use async_trait::async_trait;
+use jsonrpsee::core::RpcResult;
+use reth_network::NetworkHandle;
+use reth_network_api::NetworkInfo;
+use reth_primitives::{keccak256, Bytes, H256};
+use reth_rpc_api::Web3ApiServer;
+
+use crate::result::ToRpcResult;
+
+/// `web3` API implementation.
+///
+/// This type provides the functionality for handling `web3` related requests.
+pub struct Web3Api {
+    /// An interface to interact with the network
+    network: NetworkHandle,
+}
+
+impl Web3Api {
+    /// Creates a new instance of `Web3Api`.
+    pub fn new(network: NetworkHandle) -> Web3Api {
+        Web3Api { network }
+    }
+}
+
+#[async_trait]
+impl Web3ApiServer for Web3Api {
+    async fn client_version(&self) -> RpcResult<String> {
+        let status = self.network.network_status().await.map_internal_err(|e| e.to_string())?;
```

Comment:
was just added: https://github.com/paradigmxyz/reth/pull/993
```suggestion
        let status = self.network.network_status().await.to_rpc_result()?;
```